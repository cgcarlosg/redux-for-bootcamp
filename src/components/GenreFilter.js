import React from 'react'
import { genresMap } from '../constants'
import store from '../redux/store';
// import PropTypes from 'prop-types'


const GenreFilter = ({ changeFilter }) => {
    console.log(store.getState())
    return (
        <div>
            <select
        name="filter"
        id="filter"
 
        onChange={e => {
            changeFilter(e.target.value)
        }}
        >
        <option key="all" value={9999}>All</option>
        {Object.keys(genresMap).map(item => <option key={item} value={item}>{genresMap[item]}</option>)}
        </select>
        </div>
    )
}


// GenreFilter.propTypes = {

// }
export default GenreFilter