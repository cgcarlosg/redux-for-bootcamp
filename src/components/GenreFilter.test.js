import React from 'react';
import renderer from 'react-test-renderer';
import GenreFilter from './GenreFilter'

it('Renders Filter Options', () => {
    const popular = renderer
      .create(
        <GenreFilter handleChange={() => 'Comedy'} Value="Comedy" />,
      )
      .toJSON();
  
    expect(popular).toMatchSnapshot();
  });