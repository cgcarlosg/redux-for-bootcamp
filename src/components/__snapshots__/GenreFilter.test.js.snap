// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Filter Options 1`] = `
<div
  className="select"
>
  <select
    className="input"
    id="filter"
    name="filter"
    onChange={[Function]}
  >
    <option
      value="All"
    >
      All
    </option>
    <option
      value="Adventure"
    >
      Adventure
    </option>
    <option
      value="Fantasy"
    >
      Fantasy
    </option>
    <option
      value="Animation"
    >
      Animation
    </option>
    <option
      value="Drama"
    >
      Drama
    </option>
    <option
      value="Horror"
    >
      Horror
    </option>
    <option
      value="Action"
    >
      Action
    </option>
    <option
      value="Comedy"
    >
      Comedy
    </option>
    <option
      value="History"
    >
      History
    </option>
    <option
      value="Western"
    >
      Western
    </option>
    <option
      value="Thriller"
    >
      Thriller
    </option>
    <option
      value="Crime"
    >
      Crime
    </option>
    <option
      value="Documentary"
    >
      Documentary
    </option>
    <option
      value="Science Fiction"
    >
      Science Fiction
    </option>
    <option
      value="Mystery"
    >
      Mystery
    </option>
    <option
      value="Music"
    >
      Music
    </option>
    <option
      value="Romance"
    >
      Romance
    </option>
    <option
      value="Family"
    >
      Family
    </option>
    <option
      value="War"
    >
      War
    </option>
    <option
      value="TV Movie"
    >
      TV Movie
    </option>
  </select>
</div>
`;
